karakeep: 0.24.1->0.25.0

tetov opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

tetov committed:
karakeep: 0.24.1->0.25.0

tetov commented:
```
> nix flake prefetch github:karakeep-app/karakeep/v0.25.0
Downloaded 'github:karakeep-app/karakeep/5b520667e8c0cc9003234611b3f26bb156f1a20a' to '/nix/store/c3j27xi8i9m3nw2b4q5k7x6d7gkcl59z-source' (hash 'sha256-eAiRvesUZIwTaj7CSxtI4rkGlTkgVjbzzwjYaXlhSuo=').
```

JuliusFreudenberger commented:
> This builds (locally) without changing the hash, so something is wrong..

Then you specify the new version but still fetch the sources for the old version from cache.nixos.org

You can specify an empty string as hash and will get the new hash in an error message during build.

tetov committed:
karakeep: add update script and version check

JuliusFreudenberger commented:
I'm also not an expert in this but my understanding is as follows: If something with this hash is present it will be used, no matter if the versions match. So you end up specifying a new version while still building the old one.
That's why something like `versionCheckHook` helps discover issues like this.

tetov commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 416487`
Commit: `0088414ab44239e4e8a06884475cf5750e50fae0`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>karakeep</li>
  </ul>
</details>

tetov commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 416487`
Commit: `1d6dab032b332b99c03882ea0de099485b7bd39b`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>karakeep</li>
  </ul>
</details>

HritwikSinghal commented:
> This builds (locally) without changing the hash, so something is wrong..

> This builds (locally) without changing the hash, so something is wrong..

see https://discourse.nixos.org/t/why-wouldnt-the-sha256-hash-change-when-fetching-from-a-different-url-when-updating-a-package/11745/3

i encountered the same issue on my local machine when i upgraded my local vuetorrent pkg (i have a local one which use prebuild binary instead of building from source). I would update only the `version` field (which i see you did in [this](https://github.com/NixOS/nixpkgs/commit/f07fd9cad5bbe5b5f951e0270654162e96adc9b4) old commit). It should intuatively download the new binary and complain of hash mismatch, but would build without failing the hash.

i think replacing the hash with `lib.fakeHash` fixed it and it threw that error, thus i was able to get the correct hash. I don't remember now exactly how i fixed it

Edit: i can see you already fixed it. What was your solution?

JuliusFreudenberger commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 416487`

Logs: https://github.com/JuliusFreudenberger/nixpkgs-review-gha/actions/runs/15642759732


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>karakeep</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>karakeep</li>
  </ul>
</details>

tetov commented:
@HritwikSinghal: I updated the hash based on my `nix flake prefetch` result and the pnpmDeps just by running it with lib.fakeHash. An empty (or just different) hash would have worked too. 

I added update script (after checking that it could deal with pnpmDeps), and a testVersion. But that doesn't help much when the CLI reports the wrong version :(. Filed a bug report upstream.
